File: Cpp/Sort Letters by Case.cpp
class Solution {
public:
    /**
     * @param chars: The letters array you should sort.
     */
    bool isupper(char ch) {
        if (ch >= 'a' && ch <= 'z') return false;
        else return true;
    }
    void sortLetters(string &letters) {
        // write your code here
        int left = 0, right = letters.size() - 1;
        while (left < right) {
            while (left < right && !isupper(letters[left])) {
                left ++;
            }
            while (left < right && isupper(letters[right])) {
                right --;
            }
            if (left < right) {
                char ch = letters[left];
                letters[left] = letters[right];
                letters[right] = ch;
            }
        }
    }
};

File: Java/Sort-Letters-by-Case.java
public class Solution {
    /**
     *@param chars: The letter array you should sort by Case
     *@return: void
     */
    public boolean isupper(char ch) {
        if (ch >= 'a' && ch <= 'z') return false;
        else return true;
    }
    public void sortLetters(char[] chars) {
        //write your code here
        int left = 0, right = chars.length - 1;
        while (left < right) {
            while (left < right && !isupper(chars[left])) {
                left ++;
            }
            while (left < right && isupper(chars[right])) {
                right --;
            }
            if (left < right) {
                char temp = chars[left];
                chars[left] = chars[right];
                chars[right] = temp;
            }
        }
    }
}

